src/java.base/share/classes/jdk/internal/loader/NativeLibraries.java

```diff
@@ -487,7 +424,7 @@ public int getCounter() {
     private static final Map<String, CountedLock> nativeLibraryLockMap =
             new ConcurrentHashMap<>();
 
-    private static void acquireNativeLibraryLock(String libraryName) {
+    static void acquireNativeLibraryLock(String libraryName) {
```

Comment:
Note sure about these two routines. Should they be shared? Should we have two locks, one for JNI, one for raw? Of course, as is the code looks fine - but I wonder if a single lock isn't overly strict.

Reply:
I initially planned to come back to this but I obviously forgot.  

This does not need the per-library locking for JNI library which avoids the deadlock due to the class and library loading scenarios involving JNI_OnLoad, class initialization, and `Sytem::loadLibrary`.   I now removed the locking entirely as it uses the concurrent hash map to register/unregister the loaded libraries.  Much simplified.